app-backend/tool/src/main/scala/eval/Interpreter.scala

```diff
@@ -99,19 +100,20 @@ object Interpreter extends LazyLogging {
     */
   def interpretGlobal(
     ast: MapAlgebraAST,
+    params: EvalParams,
     source: RFMLRaster => Future[Option[Tile]]
   )(implicit ec: ExecutionContext): Future[Interpreted] = {
 
     def eval(ast: MapAlgebraAST): Future[Interpreted] = ast match {
-      case RFMLRasterSource(id, label, None) =>
-        logger.debug(s"case unbound rastersource at $id")
-        Future.successful { Invalid(NonEmptyList.of(MissingParameter(id))) }
-
-      case RFMLRasterSource(id, label, Some(ref)) =>
-        logger.debug(s"case bound rastersource at $id")
-        source(ref).map { maybeTile =>
-          val maybeLT = maybeTile.map(LazyTile.apply)
-          Validated.fromOption(maybeLT, { NonEmptyList.of(RasterRetrievalError(id, ref.id)) })
+      case Source(id, label) =>
+        if (params.sources.isDefinedAt(id)) {
+          val rfmlRaster = params.sources(id)
```

Comment:
Ah they're assumed to be `RFMLRaster`s in all cases... and the function to get them is passed in. Cool.